index.js

```diff
@@ -1,73 +1,105 @@
-const express = require('express');
-const path = require('path');
-const helmet = require('helmet')
+const express = require("express");
+const path = require("path");
+const helmet = require("helmet");
+const url = require("url");
 
 const app = express();
-const NEXCHANGE_ROOT = process.env.NEXCHANGE_ROOT
-const ICO_ROOT = process.env.ICO_ROOT
+const NEXCHANGE_ROOT = process.env.NEXCHANGE_ROOT;
+const ICO_ROOT = process.env.ICO_ROOT;
 
 //Helmet helps you secure your Express apps by setting various HTTP headers.
-app.use(helmet())
-app.use('/ico', express.static(path.resolve(ICO_ROOT)));
-app.use(express.static(path.resolve(NEXCHANGE_ROOT), {index: false}));
+app.use(helmet());
+app.use("/ico", express.static(path.resolve(ICO_ROOT)));
+app.use(express.static(path.resolve(NEXCHANGE_ROOT), { index: false }));
 
 function getCur(qParam) {
-    return qParam.toUpperCase().substr(-3);
+  return qParam.toUpperCase().substr(-3);
 }
 
-app.get('/ico', (req, res) => {
-  res.sendFile(path.resolve(ICO_ROOT, 'index.html'));
+app.get("/ico", (req, res) => {
+  res.sendFile(path.resolve(ICO_ROOT, "index.html"));
 });
 
 // Handling lowercase order Ids
-var orderUppercase = (req, res, next) => {
-
-  if (req.params.orderId) {
-    const orderIdUP =  req.params.orderId.toUpperCase()
-    if (req.params.orderId != orderIdUP) {
-      res.redirect('/order/' + orderIdUP)
-    }
-    else {
-      next()
-    }
+const orderUppercase = (req, res, next) => {
+  const orderIdUP = req.params.orderId.toUpperCase();
+  if (req.params.orderId !== orderIdUP) {
+    res.redirect(`/${lang}/order/${orderIdUP}`);
+  } else {
+    next();
   }
 };
 
 // General handler for the rest of the URLs
-var generalHandler = (req, res) => {
+const generalHandler = (req, res) => {
+  let host = undefined;
+  let urlPath = req.path;
+  const langInPath = urlPath.split("/")[1];
+  const langInParam = req.query.lang ? req.query.lang.toLowerCase() : undefined;
+  let lang = langInPath || langInParam || "en";
+  const fromCurr = req.query.cur_from;
+  const toCurr = req.query.cur_to;
+  const validLanguages = ["en", "de", "ru"];
 
   let redirectRequired = false;
-  let params = {};
-  if (req.query.cur_from && req.query.cur_to) {
+
+  // Dont add language path in url, if its not n.exchange
```

Comment:
what if some new whitelabels come that use our lang param?

However whitelabels generally do not get traffic from bestchange

Reply:
Client side routing will be used

Reply:
Replaced it with `const languageRedirect = true;`
If true, language will be added in url path while redirecting